src/coreclr/vm/amd64/cgenamd64.cpp

```diff
@@ -206,7 +206,7 @@ void HelperMethodFrame::UpdateRegDisplay(const PREGDISPLAY pRD, bool updateFloat
 
 #endif // TARGET_UNIX
 
-#define CALLEE_SAVED_REGISTER(regname) pRD->pCurrentContextPointers->regname = m_MachState.m_Ptrs.p##regname;
+#define CALLEE_SAVED_REGISTER(regname) pRD->pCurrentContextPointers->regname = (DWORD64 *)(TADDR *)m_MachState.m_Ptrs.p##regname;
```

Comment:
```suggestion
#define CALLEE_SAVED_REGISTER(regname) pRD->pCurrentContextPointers->regname = (DWORD64 *)m_MachState.m_Ptrs.p##regname;
```
Is this enough?

Reply:
Nope, doesn’t work. It’s typed as the `__DPtr` class and it doesn’t have the cast operator to make that work.

Reply:
osx-x64 with single cast:
```
  /Users/filipnavara/Projects/runtime/src/coreclr/vm/amd64/cgenamd64.cpp:210:5: error: cannot cast from type 'PTR_TADDR' (aka '__DPtr<unsigned long>') to pointer type 'DWORD64 *' (aka 'unsigned long long *')
      ENUM_CALLEE_SAVED_REGISTERS();
      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /Users/filipnavara/Projects/runtime/src/coreclr/vm/amd64/cgencpu.h:182:5: note: expanded from macro 'ENUM_CALLEE_SAVED_REGISTERS'
      CALLEE_SAVED_REGISTER(R12) \
      ^~~~~~~~~~~~~~~~~~~~~~~~~~
  /Users/filipnavara/Projects/runtime/src/coreclr/vm/amd64/cgenamd64.cpp:209:80: note: expanded from macro 'CALLEE_SAVED_REGISTER'
```